### PR TITLE
dnscrypt-proxy: new package at 2.0.45

### PR DESCRIPTION
**Pull request by git-bruh**
## Description of package

A flexible DNS proxy, with support for encrypted DNS protocols

## Installed manifest of package

```
/var/db/kiss/installed/dnscrypt-proxy/version
/var/db/kiss/installed/dnscrypt-proxy/sources
/var/db/kiss/installed/dnscrypt-proxy/manifest
/var/db/kiss/installed/dnscrypt-proxy/files/dnscrypt-proxy.run
/var/db/kiss/installed/dnscrypt-proxy/files/
/var/db/kiss/installed/dnscrypt-proxy/etcsums
/var/db/kiss/installed/dnscrypt-proxy/depends
/var/db/kiss/installed/dnscrypt-proxy/checksums
/var/db/kiss/installed/dnscrypt-proxy/build
/var/db/kiss/installed/dnscrypt-proxy/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/bin/dnscrypt-proxy
/usr/bin/
/usr/
/etc/sv/dnscrypt-proxy/supervise
/etc/sv/dnscrypt-proxy/run
/etc/sv/dnscrypt-proxy/
/etc/sv/
/etc/dnscrypt-proxy/forwarding-rules.txt
/etc/dnscrypt-proxy/dnscrypt-proxy.toml
/etc/dnscrypt-proxy/cloaking-rules.txt
/etc/dnscrypt-proxy/captive-portals.txt
/etc/dnscrypt-proxy/blocked-names.txt
/etc/dnscrypt-proxy/blocked-ips.txt
/etc/dnscrypt-proxy/allowed-names.txt
/etc/dnscrypt-proxy/allowed-ips.txt
/etc/dnscrypt-proxy/
/etc/
```

## New package

- [X] Latest upstream version.
- [X] I agree to maintain this package.
- [X] I agree to notify the @kiss-community maintainers if I can no longer maintain this package.

### CODE REVIEW

**Comment by illiliti** [comment]
Use `go build -mod=vendor` to allow offline building.

File: community/dnscrypt-proxy/build [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh -e
+
+cd dnscrypt-proxy
+
+export GOPATH="$PWD/go"
+export GO111MODULE=on
```
Comment (illiliti) [inline community/dnscrypt-proxy/build:6]: Why?

File: community/dnscrypt-proxy/build [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+go build -mod=vendor
+
+install -Dm755 dnscrypt-proxy "$1/usr/bin/dnscrypt-proxy"
+install -Dm644 example-dnscrypt-proxy.toml "$1/etc/dnscrypt-proxy/dnscrypt-proxy.toml"
+
+for config in ./*.txt; do
+    install -Dm644 "$config" "$1/etc/dnscrypt-proxy/$(echo "$config" | sed 's/^.\{10\}//')"
+done
```
Comment (illiliti) [inline community/dnscrypt-proxy/build:15]:
```
for f in *.txt; do
    install -Dm644 "$f" "$1/etc/dnscrypt-proxy/${f#example-}"
done
```

**Review by illiliti — CHANGES_REQUESTED** [review]

File: community/dnscrypt-proxy/build (hunk 1 above)
Comment (git-bruh) [reply community/dnscrypt-proxy/build:6]: exporting `GOPATH` won't put `go`'s stuff in `$HOME/go`, saw both these variables used in other golang packages aswell so added them.

Comment (illiliti) [reply community/dnscrypt-proxy/build]: These variables are no longer needed. `-mod=vendor` prevents cluttering $HOME

Comment (sdsddsd1) [reply community/dnscrypt-proxy/build]: The style guide could then be updated.
https://github.com/kiss-community/wiki/blob/master/kiss/style-guide.txt#L334

Comment (dilyn-corner) [reply community/dnscrypt-proxy/build]: Indeed it could. I'll make that adjustment soon(tm).

Comment (sdsddsd1) [reply community/dnscrypt-proxy/build]: https://github.com/kiss-community/wiki/pull/7

**Pull request merged**